api/data_explorer/__main__.py

```diff
@@ -212,38 +249,35 @@ def _process_facets(es):
 
     for facet_config in facets_config:
         es_base_field_name = facet_config['elasticsearch_field_name']
+        es_parent_field_name = es_base_field_name.rsplit('.', 1)[0]
         is_time_series = elasticsearch_util.is_time_series(
-            es, es_base_field_name, mapping)
+            es_base_field_name, mapping)
+        parent_is_time_series = elasticsearch_util.is_time_series(
+            es_parent_field_name, mapping)
+
         if is_time_series:
             time_series_vals = elasticsearch_util.get_time_series_vals(
-                es, es_base_field_name, mapping)
+                es_base_field_name, mapping)
             es_field_names = [
                 es_base_field_name + '.' + tsv for tsv in time_series_vals
             ]
+            for es_field_name in es_field_names:
+                separate_panel = (es_field_name in facets
+                                  and facets[es_field_name]['separate_panel'])
+                _add_facet(es_field_name, is_time_series, True, separate_panel,
+                           facet_config, time_series_vals, facets, es, mapping)
+        elif parent_is_time_series:
+            time_series_vals = elasticsearch_util.get_time_series_vals(
+                es_parent_field_name, mapping)
+            time_series_panel = (
+                es_base_field_name in facets
+                and facets[es_base_field_name]['time_series_panel'])
+            _add_facet(es_base_field_name, is_time_series, time_series_panel,
+                       True, facet_config, time_series_vals, facets, es,
+                       mapping)
         else:
-            time_series_vals = []
-            es_field_names = [es_base_field_name]
-
-        for es_field_name in es_field_names:
-            if es_field_name in facets:
-                raise EnvironmentError('%s appears more than once in ui.json' %
-                                       es_field_name)
-            field_type = elasticsearch_util.get_field_type(
-                es, es_field_name, mapping)
-            ui_facet_name = facet_config['ui_facet_name']
-            if es_field_name.startswith('samples.'):
-                ui_facet_name = '%s (samples)' % ui_facet_name
-            facets[es_field_name] = {
-                'ui_facet_name': ui_facet_name,
-                'type': field_type,
-                'is_time_series': is_time_series
-            }
-            if 'ui_facet_description' in facet_config:
-                facets[es_field_name]['description'] = facet_config[
-                    'ui_facet_description']
-            facets[es_field_name][
-                'es_facet'] = elasticsearch_util.get_elasticsearch_facet(
-                    es, es_field_name, field_type, time_series_vals)
```

Comment:
Can you update 282? By convention we document the things in app.app.config somewhere in this method. Eg I think time_series_panel is entries in the dict.